### ISSUE TITLE
Why is the _id on MongoDB inconsistent with the API query id? 

### ISSUE THREAD

**Issue by SpongeKevin** [abpframework/abp#4827]
## Problem Objective:
I want to know if there's something wrong with me, or if there's an implication.

Because if I do a query in [Book Store: Razor Pages UI & Entity Framework Core](https://github.com/abpframework/abp-samples/tree/master/BookStore), I get the same id that matches the database.

## Environment:
1. BookStore-Angular-MongoDb(commit:033bc32)
2. MongoDB 4.2
3. Mongodb client:Robo 3T

## Problem Details:
Using the Sample Applications (Book Store Angular UI & MongoDB) 
found that the id from api(/api/app/book) is not the same as the _id in MongoDB.

### Step2: Run DbMigrator and start HttpApi.Host then get data from /api/app/book， i get
```
{
  "totalCount": 2,
  "items": [    
    {
      "name": "The Hitchhiker's Guide to the Galaxy",
      "type": 7,
      "publishDate": "1995-09-26T16:00:00Z",
      "price": 42,
      "lastModificationTime": null,
      "lastModifierId": null,
      "creationTime": "2020-07-18T15:34:58.43Z",
      "creatorId": null,
      "id": "90be5d55-59b9-13db-1572-39f674bd427e"
    },
    {
      "name": "1984",
      "type": 3,
      "publishDate": "1949-06-07T16:00:00Z",
      "price": 19.84,
      "lastModificationTime": null,
      "lastModifierId": null,
      "creationTime": "2020-07-18T15:34:58.337Z",
      "creatorId": null,
      "id": "23fff07c-7977-b055-6309-39f674bd4221"
    }
  ]
}
```

### Step3: Using mongodb client and select the collection(Books)，i get 
```
// 1
{
    "_id": UUID("7cf0ff23-7779-55b0-6309-39f674bd4221"),
    "ConcurrencyStamp": "bdac8fc24a2044afa9fcda5ef6272c92",
    "CreationTime": ISODate("2020-07-18T15:34:58.337Z"),
    "CreatorId": null,
    "LastModificationTime": null,
    "LastModifierId": null,
    "Name": "1984",
    "Type": NumberInt("3"),
    "PublishDate": ISODate("1949-06-07T16:00:00.000Z"),
    "Price": 19.8400001525879
}

// 2
{
    "_id": UUID("555dbe90-b959-db13-1572-39f674bd427e"),
    "ConcurrencyStamp": "178e36ce3c504a6cb6b37ff3dd741998",
    "CreationTime": ISODate("2020-07-18T15:34:58.43Z"),
    "CreatorId": null,
    "LastModificationTime": null,
    "LastModifierId": null,
    "Name": "The Hitchhiker's Guide to the Galaxy",
    "Type": NumberInt("7"),
    "PublishDate": ISODate("1995-09-26T16:00:00.000Z"),
    "Price": 42
}
```

### Step4: By comparing the ID of I get by different method, I have found out that they are a little different between the sequence.

```
Name: The Hitchhiker's Guide to the Galaxy

Id from api:90be5d55-59b9-13db-1572-39f674bd427e
Id from mongoDB:555dbe90-b959-db13-1572-39f674bd427e
```
Id from these three blocks are reversed.

**Comment by berkansasmaz** [comment]
As I understand the problem is not caused by the ABP framework.

### HttpApi.Host then get data from /api/app/book
<img width="457" alt="Screen Shot 2020-07-21 at 23 30 39" src="https://user-images.githubusercontent.com/31216880/88104306-ed0aac80-cbaa-11ea-907f-9e41f7151997.png">

### Using mongodb client and select the collection(Books)
<img width="942" alt="Screen Shot 2020-07-21 at 23 34 05" src="https://user-images.githubusercontent.com/31216880/88104383-07dd2100-cbab-11ea-9ac7-fa71f5da6638.png">

### The step I took for the solution:
<img width="1440" alt="Screen Shot 2020-07-21 at 23 33 25" src="https://user-images.githubusercontent.com/31216880/88104497-2f33ee00-cbab-11ea-8e8a-bfe32171c288.png">

**Comment by SpongeKevin** [comment]
Thank you, I saw your solution and realized that my driver is the java version.
After I changed to the c# version, the id are the same as i got from api.

Then i found this [article](https://studio3t.com/knowledge-base/articles/mongodb-best-practices-uuid-data/).  
It say the same UUID differs depending on the platform used to access it.

![image](https://user-images.githubusercontent.com/35734121/88119679-c2911180-cbf3-11ea-8952-e569603a3c9f.png)

So the problem is my driver, I need to look deeper into mongodb.
